client/src/PublicRequest.ts

```diff
@@ -173,14 +171,14 @@ export type Result = RedirectResult | IFrameResult;
 
 export type ResultType<T extends RedirectRequest> =
     T extends Is<T, SignMessageRequest> | Is<T, SignTransactionRequest> ? SignatureResult :
-    T extends Is<T, DeriveAddressRequest> ? Is<T, DeriveAddressResult> :
+    T extends Is<T, DeriveAddressRequest> ? Is<T, DerivedAddress[]> :
```

Comment:
```suggestion
    T extends Is<T, DeriveAddressRequest> ? DerivedAddress[] :
```
That was my fault, but good opportunity to sneak in and fix it.